pkg/phlaredb/symdb/stacktrace_tree_test.go

```diff
@@ -220,9 +222,48 @@ func Benchmark_stacktrace_tree_insert(b *testing.B) {
 	b.ReportAllocs()
 
 	for i := 0; i < b.N; i++ {
-		x := newStacktraceTree(0)
+		x := newStacktraceTree(defaultStacktraceTreeSize)
 		for j := range p.Sample {
 			x.insert(p.Sample[j].LocationId)
 		}
 	}
 }
+
+func Benchmark_stacktrace_tree_insert_default_sizes(b *testing.B) {
+	if os.Getenv("COMPARE_STACKTRACETREE_INSERT_DEFAULT_SIZES") != "true" {
+		b.Skip("This benchmark is only used for comparing different initial tree sizes")
+	}
+
+	p, err := pprof.OpenFile("testdata/profile.pb.gz")
+	require.NoError(b, err)
+
+	b.ResetTimer()
+
+	for _, size := range []int{0, 10, 1024, 2048, 4096, 8192} {
+		b.Run("size="+strconv.Itoa(size), func(b *testing.B) {
+			b.ReportAllocs()
+
+			for i := 0; i < b.N; i++ {
+				x := newStacktraceTree(size)
+				for j := range p.Sample {
+					x.insert(p.Sample[j].LocationId)
+				}
+				// nodes = max(nodes, x.len())
+			}
+
+		})
```

Comment:
Please note that the tree does not grow over ~1.8K nodes in this specific case. 4K and 8K cases mostly benchmark Go allocator.

Reply:
Yeah, that was my impression as well for 4k and 8k, I just added there to have a comparison on overestimating capacity.